lom/src/engine/sequenceHandler.go

```diff
@@ -0,0 +1,832 @@
+ *
+ *  Note:
+ *      a.  Every action response is published, even if stale.
+ *      b.  First action of the sequence is published with state='init' and upon
+ *          sequence completion, it is re-publsihed with state="complete" and
+ *          appropriate result code.
+ *
+ *  Failure scenarios:
+ *  A.  Sequence config could be missing data, like action w/o config, the sequence
+ *      config is removed, one or more actions in the sequence is marked 'disabled'
+ *      and more
+ *  
+ *  B.  An action in the sequence could have failed result_code != 0
+ *
+ *  C.  Request or sequence could have timedout
+ *
+ *  D.  Any other fatal failure ?
+ *
+ *  On any failure:
+ *      i.  Seq is processed as complete with possible error codes
+ *      ii. TODO: Honor *mandatory* flag, as call 
+ */
+
+
+/*
+ * Active request - Requests sent out awaiting response with or w/o timeout
+ */
+type activeRequest_t struct {
+    req         *ServerRequestData
+    anomalyID   string
+    reqExpEpoch int64       /* Expiry time. A value of 0 means no expiry */
+}
+
+/*
+ * Action vs request -- All active requests
+ *  A request may or may not have a timeout.
+ *
+ * NOTE: Only one outstanding request per Action
+ *       Hence can be keyed off by action.
+ *
+ * NOTE: Engine may abort a sequence when its current request does not
+ *       respond within its timeout. But it does *not* remove request from
+ *       the active list, until the action/client sends the response. This
+ *       is because, it implies that the plugin/action is still busy with
+ *       last given request and we can't raise another request, as there 
+ *       can be atmost one active request per action.
+ *
+ *       In case of any fatal failure, a de-register action/client or 
+ *       re-register by client will drop the requests associated with 
+ *       de-registered actions.
+ */
+type activeRequestsList_t map[string]*activeRequest_t
+
+
+/*
+ * Sequence is kicked off when first request of a sequence responds.
+ *
+ * There can be many sequences. But only one sequence can be active.
+ *
+ * SequenceInfo and related data is read & filled upon sequence creation
+ * which is when the first action of the sequence responds.
+ * 
+ * Only one sequence can be active. Any sequence created when one is active
+ * is added to pending Q. Going active implies invoking other actions of the
+ * sequence, sequentially. Sequence remains active until all actions of the
+ * sequence are complete or timeout, whichever comes earlier.
+ *
+ * When there are multiple sequences ready to go active, they are first sorted
+ * by Priority and then by order of arrival within same priority
+ *
+ * All sequences have timeout. Some sequence may timeout, even before going 
+ * active.
+ *
+ */
+
+type sequenceStatus_t int
+const (
+    sequenceStatus_pending = sequenceStatus_t(iota)
+    sequenceStatus_running
+    sequenceStatus_complete
+)
+
+
+type sequenceState_t struct {
```

Comment:
Suggestion to remove _t from everywhere.

Reply:
I think it is fine for now, but for later code, could we avoid this ? I think _t is not needed to differentiate type vs var.

Reply:
We need a distinct way to differentiate type declaration vs var.
If you have any idea, please share.